src/core/meshcop/commissioner.cpp

```diff
@@ -706,15 +715,15 @@ void Commissioner::HandleLeaderPetitionResponse(Coap::Message *         aMessage
     CommissionerSessionIdTlv sessionId;
     bool                     retransmit = false;
 
-    VerifyOrExit(mState == OT_COMMISSIONER_STATE_PETITION, SetState(OT_COMMISSIONER_STATE_DISABLED));
+    VerifyOrExit(mState == OT_COMMISSIONER_STATE_PETITION, Stop(false));
```

Comment:
Why should the commissioner stop itself if it received a `COMM_PET.rsp` in an invalid state? Why not ignore the unexpected `COMM_PET.rsp`?

Reply:
I agree with what you said. The current code is same as it was.